### ISSUE TITLE
Environment variable expansion problems

### ISSUE THREAD

**Issue by kkobashi** [lxc/lxd#1552]
Problem #1:
Setting environment variables inside rootfs/etc/environment doesn't work. The file appears to never get read.

Problem #2:
Due to Problem #1 issue, tried to set the environment property inside a LXD profile. For example: 

```
lxc profile set p1 environment.JAVA_HOME /java/jdk1.8.0_72
lxc profile set p1 environment.PATH $PATH:$JAVA_HOME/bin
```

After lxc exec -- /bin/bash into the container, I do an env to find that environment variable expansion does not work.

lxc profile show p1 result:

```
environment.JAVA_HOME: /java /jdk1.8.0_72
environment.PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/bin
```

**Comment by nehaljwani** [comment]
You have to apply the profile to the container. Following works for me:
```
[root@lxc-dev ~]# lxc profile create customProfile
Profile customProfile created
[root@lxc-dev ~]# lxc profile set customProfile environment.JAVA_HOME /java/jdk1.8.0_72                  
[root@lxc-dev ~]# lxc profile show  customProfile
name: customProfile
config:
  environment.JAVA_HOME: /java/jdk1.8.0_72
devices: {}
[root@lxc-dev ~]# lxc profile apply ubuntu customProfile
Profile customProfile applied to ubuntu
[root@lxc-dev ~]# lxc start ubuntu
[root@lxc-dev ~]# lxc exec ubuntu -- bash
root@ubuntu:~# env |grep -i java
JAVA_HOME=/java/jdk1.8.0_72
root@ubuntu:~# 

```

**Comment by stgraber** [comment]
That's perfectly normal, LXD can't expand variables for you. It will just set whatever value you pass it to the environment variable, but that's it.

There's also nothing we can do about this, LXD isn't aware of the environment the shell in the container will be giving you before said shell is executed.

**Comment by kkobashi** [comment]
nehaljwani:
I did apply the profile. That is how I got JAVA_HOME set in the result.

stgraber: 
I played around with this more. There's actually a solution to this that doesn't involve having to set lxc environment.*

 /etc/environment only **assigns** system wide environment variables. 

One needs to **export** them so they are passed through to child processes (i.e. lxc bash shell).

`lxc exec -- /bin/bash`

Since this launches an **interactive, non-login shell**, the solution is to place all environment variables inside each container's **/rootfs/etc/bash.bashrc** like so:

```
export JAVA_VERSION=1.8.0_72
export JAVA_HOME=/java/jdk$JAVA_VERSION
export PATH=$PATH:$JAVA_HOME/bin

```

Which works natively inside container c1 with environment variable expansion:

```
root@UbuntuLXD1:/# lxc exec c1 -- /bin/bash
root@c1:~# env
(snip)
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/java/jdk1.8.0_72/bin
JAVA_HOME=/java/jdk1.8.0_72
JAVA_VERSION=1.8.0_72
container=lxc
(snip)
_=/usr/bin/env

```
